### PR TITLE
applier: make SSLError retryable

### PR DESCRIPTION
**Pull request by locker**
A certificate used for an encrypted replication connection may expire, in which case the user will be forced to reconfigure the connection both on the master and on the replica to use the new certificate. If the replica completes the reconfiguration first, it will fail to connect to the master and never retry, even though the master should complete its reconfiguration in a moment.

To avoid that, we should make the applier retry to connect on `SSLError`, just like it does on `SocketError`.

Needed for https://github.com/tarantool/tarantool-ee/issues/107
Needed for 2.10

A test is added in the EE PR: https://github.com/tarantool/tarantool-ee/pull/125

### CODE REVIEW

File: src/box/errcode.h [hunk 1]
```diff
@@ -295,6 +295,7 @@ struct errcode_record {
 	/*240 */_(ER_COMPLEX_FOREIGN_KEY_FAILED, "Foreign key constraint '%s' failed: %s") \
 	/*241 */_(ER_WRONG_SPACE_UPGRADE_OPTIONS, "Wrong space upgrade options: %s") \
 	/*242 */_(ER_NO_ELECTION_QUORUM,	"Not enough peers connected to start elections: %d out of minimal required %d")\
+	/*243 */_(ER_SSL,			"%s") \
```
Comment (Gerold103) [inline src/box/errcode.h:298]: Why do you need the new code? I don't see it used in this patch neither in the EE patch.

Comment (locker) [reply src/box/errcode.h:298]: Applier uses `box_error_code()` for better logging. In particular, it [logs](https://github.com/tarantool/tarantool/blob/73733e51331027424ddce117c09402b7d61db490/src/box/applier.cc#L115-L133) "will retry every X seconds" only on retryable error while `ER_PROC_LUA`, which is currently returned by `box_error_code()` for `SSLError`, isn't retryable. I moved the code adding the new error code to a separate commit and added a test case to `unit/error.test` checking `box_error_code()`.

**Review by Gerold103 — APPROVED** [review]
Thanks for the fixes, LGTM!

**Comment by locker** [comment]
Cherry-picked for 2.10.

**Pull request merged**